pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -892,6 +897,148 @@ var _ = Describe("Snapshotter", func() {
 				})
 			})
 		})
+
+		Describe("Scenarios to update full snapshot lease", func() {
+			var (
+				ssr   *Snapshotter
+				lease *v1.Lease
+			)
+			BeforeEach(func() {
+				healthConfig.SnapshotLeaseRenewalEnabled = true
+				snapstoreConfig = &brtypes.SnapstoreConfig{Container: path.Join(outputDir, "default.bkp")}
+				store, err = snapstore.GetSnapstore(snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(os.Setenv("POD_NAME", "test_pod")).To(Succeed())
+				Expect(os.Setenv("POD_NAMESPACE", "test_namespace")).To(Succeed())
+				os.Setenv("UNIT_TEST", "true")
+
+				lease = &v1.Lease{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Lease",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      brtypes.DefaultFullSnapshotLeaseName,
+						Namespace: os.Getenv("POD_NAMESPACE"),
+					},
+				}
+
+			})
+			AfterEach(func() {
+				Expect(os.Unsetenv("POD_NAME")).To(Succeed())
+				Expect(os.Unsetenv("POD_NAMESPACE")).To(Succeed())
+				Expect(os.Unsetenv("UNIT_TEST")).To(Succeed())
+			})
+			Context("Without previous full snapshot", func() {
+				It("cannot update the lease", func() {
+					snapshotterConfig := &brtypes.SnapshotterConfig{
+						FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", 0, 0),
+					}
+					ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					ssr.PrevFullSnapshot = nil
+					ssr.FullSnapshotLeaseStopCh = make(chan struct{})
+					ssr.K8sClientset = fake.NewClientBuilder().Build()
+
+					ctx, cancel := context.WithCancel(context.Background())
+					defer cancel()
+					err := ssr.K8sClientset.Create(ctx, lease)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					ssr.SetFullSnapshotLeaseUpdatePeriod(2 * time.Second)
+					go ssr.RenewFullSnapshotLeasePeriodically()
+					time.Sleep(2 * time.Second)
+					ssr.FullSnapshotLeaseStopCh <- struct{}{}
+
+					l := &v1.Lease{}
+					Expect(ssr.K8sClientset.Get(ctx, client.ObjectKey{
+						Namespace: lease.Namespace,
+						Name:      lease.Name,
+					}, l)).To(Succeed())
+					Expect(l.Spec.HolderIdentity).To(BeNil())
+				})
+			})
+			Context("With previous full snapshot", func() {
+				It("is able to fetch and update the lease", func() {
+					snapshotterConfig := &brtypes.SnapshotterConfig{
+						FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", 0, 0),
+					}
+					ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					prevFullSnap := &brtypes.Snapshot{
+						Kind:          brtypes.SnapshotKindFull,
+						CreatedOn:     time.Now(),
+						StartRevision: 0,
+						LastRevision:  123,
+					}
+					prevFullSnap.GenerateSnapshotName()
+					ssr.PrevFullSnapshot = prevFullSnap
+					ssr.FullSnapshotLeaseStopCh = make(chan struct{})
```

Comment:
While calling func `NewSnapshotter()`, this stop channel is already getting initialised then why you to initialised it seprately again, I guess you can remove this line.